### PR TITLE
Add override to distributed tracing helper in trace_utils

### PR DESCRIPTION
**Pull request by Yun-Kim**
## Description
From #2216, it was found that several integrations (including flask.middlewares, cherrypy.middleware, falcon.middleware, tornado.handler, pyramid.trace, bottle.trace, pylons.middleware) have unique setting objects to check for the distributed_tracing setting, which are currently not compatible with the implementation in the trace_utils helper. Therefore the helper function in trace_utils.py was modified to include an override for these unique integration patches. Two test cases were added to cover the override use cases.

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: ddtrace/contrib/trace_utils.py [hunk 1]
```diff
@@ -194,13 +194,20 @@ def set_http_meta(
         span._set_str_tag(http.RETRIES_REMAIN, retries_remain)
 
 
-def activate_distributed_headers(tracer, int_config=None, request_headers=None):
+def activate_distributed_headers(tracer, int_config=None, request_headers=None, override_distributed_tracing=None):
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/trace_utils.py:197]:
```suggestion
def activate_distributed_headers(tracer, int_config=None, request_headers=None, override=None):
```

I think we can shorten this for usability :)

**Review by Kyle-Verhoog — COMMENTED** [review]
LGTM just a nit on the kwarg name

File: ddtrace/contrib/trace_utils.py [hunk 2]
```diff
@@ -194,13 +199,19 @@ def set_http_meta(
         span._set_str_tag(http.RETRIES_REMAIN, str(retries_remain))
 
 
-def activate_distributed_headers(tracer, int_config=None, request_headers=None):
+def activate_distributed_headers(tracer, int_config=None, request_headers=None, override=None):
+    # type: (Tracer, Optional[Dict[str, Any]], Optional[Dict[str, str]], Optional[bool]) -> None
     """
     Helper for activating a distributed trace headers' context if enabled in integration config.
+    int_config will be used to check if distributed trace headers context will be activated, but
+    override_distributed_tracing will override if passed any value other than None.
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/trace_utils.py:207]:
```suggestion
    the override argument will override if passed any value other than None.
```

**Review by Kyle-Verhoog — COMMENTED** [review]
just a typo but otherwise lgtm

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by P403n1x87 — APPROVED** [review]

**Pull request merged**